lib/AST/Expr.cpp

```diff
@@ -1068,6 +1068,22 @@ Type OverloadSetRefExpr::getBaseType() const {
   llvm_unreachable("Unhandled overloaded set reference expression");
 }
 
+bool ApplyExpr::shouldApplyDistributedThunk() const {
+  // only a distributed decl has a chance of needing to be invoked as a thunk
+  auto func = dyn_cast<AbstractFunctionDecl>(getCalledValue());
```

Comment:
This needs to use `dyn_cast_or_null`, which is why the macOS build failed.

Reply:
Ah I see, thank you